### DISCUSSION TITLE
Using Apalis Jobs /w Existing postgres DB

### DISCUSSION THREAD

**Discussion by nMessage** [apalis-dev/apalis#341] [Q&A]
Hey I have an existing Database which I want to add a job que for things and use triggers and other PostgreSQL features to create jobs based of the data in the database.

Side note: If I was using something like Diesel I'm pretty sure it'd work fine. But because I'm also using sqlx, I can't also use Apalis following the examples given. Any tips?

**Comment by geofmureithi** [comment]
I am unsure what you are asking.
Apalis-sql is built on top of `sqlx` It should not be an issue. 

Regarding postgres and triggers, consider looking at `apalis.push_job`.
You can make a call to push the job by passing a namespace and a json blob.
```sql
Select apalis.push_job('apalis::Email', json_build_object('subject', 'Test apalis', 'to', 'test1@example.com', 'text', 'Lorem Ipsum'));
```

**Reply by nMessage** [reply]
Thanks for the fast response. I'm having issues specifically with sqlx migrations. Sorry for not being clear.
(Apalis having its own migrations to run in order to be used. `PostgresStorage::Setup(&pool)`) If there are already any migrations, it'll error. (Or visa versa if Apalis is ran first).

I realize it's probably not the intended use.

**Reply by geofmureithi** [reply]
Apalis sql exposes the sqlx-migrator you should be able to combine our migrations with yours using [iter](https://docs.rs/sqlx/latest/sqlx/migrate/struct.Migrator.html#method.iter).
